### PR TITLE
APR-2952 Handle exception when fetching ukrlp details in gateway

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs [hunk 1]
```diff
@@ -95,22 +115,40 @@ private async Task LookupCompaniesHouseDetails(ApplyGatewayDetails applyGatewayD
...
         {
             if (!int.TryParse(charityNumberFromUkrlp, out var charityNumber))
-            { 
-                _logger.LogError("Unable to parse charity registration number from charity number in ukrlp `{charityNumber}`", charityNumberFromUkrlp);
-                return;
+            {
+                var message =
+                    $"Unable to parse charity registration number from charity number in ukrlp: '{charityNumberFromUkrlp}'";
+                _logger.LogError(message);
+                throw new ServiceUnavailableException(message);
```
Comment (DevenKShah) [inline src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs:122]: Where is this exception being handled?

File: src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs [hunk 2]
```diff
@@ -82,8 +88,22 @@ public async Task<ApplyGatewayDetails> GetExternalApiCheckDetails(Guid applicati
...
+                        new Context());
+
+            }
+            catch(Exception ex)
+            {
+                var message = $"Unable to retrieve results from companies house for company number {companyNumber}";
+                _logger.LogError(message, ex);
+                throw new ServiceUnavailableException(message);
+            }
+
```
Comment (DevenKShah) [inline src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs:105]: Empty line

File: src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs:122]: This is an service only used by Gateway, and gateway picks up the thrown exception and sends the user to a 'service unavailable' page when it happens (see the companion branch in das-roatp-gateway)

File: src/SFA.DAS.ApplyService.Domain/Entities/ApplicationSequence.cs [hunk 3]
```diff
@@ -30,8 +30,8 @@ public class ApplicationSequenceStatus
 
     public static class SequenceId
     {
-        public static int Preamble = 0;
-        public static int Stage1 = 1;
-        public static int Stage2 = 2;
+        public static int Preamble { get; } = 0;
```
Comment (DevenKShah) [inline src/SFA.DAS.ApplyService.Domain/Entities/ApplicationSequence.cs:33]: These should be constants

**Review by DevenKShah — DISMISSED** [review]

File: src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs:104]: I;'ve taken out the empty line, and if you go to the file you can see it's been removed, so no idea why this isn't showing as 'outdated'

Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.InternalApi/Services/GatewayApiChecksService.cs:104]: Wow when I added the above comment, it now shows the fix when you view.  Fixed

**Review by raviudari01 — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**